### PR TITLE
Sped up ant_metrics.calc_corr_stats and stripped out a lot of complex…

### PR DESCRIPTION
**Pull request by AaronParsons**
…ity in ant_metrics around loading files/baselines iteratively, using flagged data, allowing nans/infs in data, etc. All tests pass.

### CODE REVIEW

File: hera_qm/ant_metrics.py [hunk 1]
```diff
@@ -66,9 +66,8 @@ def get_ant_metrics_dict():
     return metrics_dict
 
 
-def calc_corr_stats(data_sum, data_diff=None, flags=None, time_alg=np.nanmean, freq_alg=np.nanmean):
-    """Calculate correlation values for all baselines, the average cross-correlation between
-    even and
+def calc_corr_stats(data_sum, data_diff=None):
+    """For all baselines, calculate average cross-correlation between even and odd in order to identify ?
```
Comment (dstorer) [inline hera_qm/ant_metrics.py:70]:
```suggestion
    """For all baselines, calculate average cross-correlation between even and odd in order to identify dead, cross-polarized, and non-time-locked antennas.
```

File: hera_qm/tests/test_ant_metrics.py [hunk 2]
```diff
@@ -26,15 +26,10 @@ def test_calc_corr_stats():
...
+    corr_stats = ant_metrics.calc_corr_stats(data_sum,data_diff)
     assert corr_stats[(0,1,'ee')] == 1
     assert corr_stats[(0,1,'nn')] == pytest.approx(0.9578,abs=1e-2)
 
-    # test with flags
-    flags[(0,1,'nn')][1,0] = True
-    corr_stats = ant_metrics.calc_corr_stats(data_sum,data_diff,flags)
-    assert corr_stats[(0,1,'nn')] == pytest.approx(0.9457,abs=1e-2)
-
     # test no diff data
```
Comment (dstorer) [inline hera_qm/tests/test_ant_metrics.py:33]: We may want to add a test for handling data that are zeros - we probably should have already had a test for this, but I'm particularly concerned now that flag handling has been removed.

**Review by dstorer — COMMENTED** [review]
Overall this looks good - my only concern is handling of data that's all zeros. Ideally we'd write a test for that scenario.

File: hera_qm/tests/test_ant_metrics.py (hunk 2 above)
Comment (jsdillon) [reply hera_qm/tests/test_ant_metrics.py:33]: It seems that we broken the special identification of "totally dead" antennas (i.e. median 0 abs(vis)). I've added a new unit test to expose the issue and then I fixed it by undoing one of @AaronParsons's changes

**Review by jsdillon — APPROVED** [review]
I've made a few fixes, but now I'm happy with this and I've verified that it produces the same result on a sample file for the file_calibration notebook (zen.2459859.48415.sum.uvh5)

**Review by dstorer — APPROVED** [review]
Looks good to me at this point!

**Pull request merged**